### PR TITLE
feat(ctb): `PreimageOracle` large preimage proposal bonds

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Adds a minimum bond requirement for large preimage proposals. The `MIN_LPP_SIZE` is likely to be lower than we expected, and bonds act as a deterrent for malicious actors to spam proposals.

### Semantics

- Min bond size: `0.25 ETH` (spitball'd; feel free to suggest a different value.)
- Bonds are deposited into the contract in `initLPP`, as the proposal is initialized.
  - A proposal's bond is paid out to the challenger if the proposal is proven to be incorrect.
  - A proposal's bond is returned to the claimant if the proposal is finalized successfully with `squeezeLPP`.

### CODE REVIEW

**Comment by clabby** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #9570** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9570?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9570?utm_source=stack-comment).

File: packages/contracts-bedrock/src/cannon/PreimageOracle.sol [hunk 1]
```diff
@@ -738,6 +758,15 @@ contract PreimageOracle is IPreimageOracle {
         }
     }
 
+    /// @notice Pay out a proposal's bond. Reverts if the transfer fails.
+    function _payoutBond(address _claimant, uint256 _uuid, address _to) internal {
+        // Pay out the bond to the claimant.
+        uint256 bond = proposalBonds[_claimant][_uuid];
+        proposalBonds[_claimant][_uuid] = 0;
+        (bool success,) = _to.call{ value: bond }("");
+        if (!success) revert BondTransferFailed();
```
Comment (ajsutton) [inline packages/contracts-bedrock/src/cannon/PreimageOracle.sol:767]: Confirming for safety - the payout is always going to the person making the call - either to the challenger when they successfully challenge or to the original proposer when they squeeze.  So there's no way you could prevent challenges from succeeding by rejecting the bond payout or anything like that.

**Review by ajsutton — APPROVED** [review]
LGTM.

File: packages/contracts-bedrock/src/cannon/PreimageOracle.sol [hunk 2]
```diff
@@ -71,6 +73,8 @@ contract PreimageOracle is IPreimageOracle {
     /// @notice Mapping of claimants to proposal UUIDs to the timestamp of creation of the proposal as well as the
     /// challenged status.
     mapping(address => mapping(uint256 => LPPMetaData)) public proposalMetadata;
+    /// @notice Mapping of claimants to proposal UUIds to bond amounts.
```
Comment (refcell) [inline packages/contracts-bedrock/src/cannon/PreimageOracle.sol]: nit
```suggestion
    /// @notice Mapping of claimants to proposal UUIDs to bond amounts.
```

**Review by refcell — APPROVED** [review]
This looks good!

File: packages/contracts-bedrock/src/cannon/PreimageOracle.sol (hunk 1 above)
Comment (clabby) [reply packages/contracts-bedrock/src/cannon/PreimageOracle.sol:767]: Yep. We also enforce that the creators of LPPs are EOAs, meaning there shouldn't be a way for a contract that created a proposal to revert on receiving ETH, etc.

**Pull request merged**